packages/core/src/components/portal/portal.tsx

```diff
@@ -56,36 +49,51 @@ export class Portal extends React.Component<IPortalProps, IPortalState> {
     public context: IPortalContext;
     public state: IPortalState = { hasMounted: false };
 
-    private targetElement: HTMLElement;
-
-    constructor(props: IPortalProps, context: IPortalContext) {
-        super(props, context);
-        this.targetElement = document.createElement("div");
-        this.targetElement.classList.add(Classes.PORTAL);
-        if (context.blueprintPortalClassName != null) {
-            this.targetElement.classList.add(context.blueprintPortalClassName);
-        }
-    }
+    private portalElement: HTMLElement;
 
     public render() {
-        // Only render `children` once this component has mounted, so they are immediately attached to the DOM tree and
-        // can do DOM things like measuring or `autoFocus`. See long comment on componentDidMount in
-        // https://reactjs.org/docs/portals.html#event-bubbling-through-portals
-        return ReactDOM.createPortal(
-            <div {...removeNonHTMLProps(this.props)} ref={this.props.containerRef}>
-                {this.state.hasMounted ? this.props.children : null}
-            </div>,
-            this.targetElement,
-        );
+        // Only render `children` once this component has mounted in a browser environment, so they are
+        // immediately attached to the DOM tree and can do DOM things like measuring or `autoFocus`.
+        // See long comment on componentDidMount in https://reactjs.org/docs/portals.html#event-bubbling-through-portals
+        return typeof document !== "undefined" && this.state.hasMounted
```

Comment:
this is a little too clever to read as a ternary, please make it more verbose w/ `if () else`. I think spelling out the 3 cases is quite useful:

```ts
if (typeof document === "undefined") {
    return null;
} else if (!this.state.hasMounted) {
    return null;
} else {
    ...
}
```